File: folkmq/src/main/java/org/noear/folkmq/client/MqSubscription.java
package org.noear.folkmq.client;

import org.noear.folkmq.common.MqConstants;

/**
 * 订阅记录
 *
 * @author noear
 * @since 1.0
 */
public class MqSubscription implements MqConsumeHandler {
    private final String topic;
    private final String consumerGroup;
    private final String queueName;
    private final boolean autoAck;
    private final MqConsumeHandler consumeHandler;


    /**
     * 主题
     */
    public String getTopic() {
        return topic;
    }

    /**
     * 消费者组
     */
    public String getConsumerGroup() {
        return consumerGroup;
    }

    /**
     * 是否自动回执
     */
    public boolean isAutoAck() {
        return autoAck;
    }

    /**
     * 相关队列名
     */
    public String getQueueName() {
        return queueName;
    }

    /**
     * 消费处理器
     */
    public MqConsumeHandler getConsumeHandler() {
        return consumeHandler;
    }

    /**
     * @param topic          主题
     * @param consumerGroup  消费者组
     * @param consumeHandler 消费处理器
     */
    public MqSubscription(String topic, String consumerGroup, boolean autoAck, MqConsumeHandler consumeHandler) {
        this.topic = topic;
        this.consumerGroup = consumerGroup;
        this.autoAck = autoAck;
        this.queueName = topic + MqConstants.SEPARATOR_TOPIC_CONSUMER_GROUP + consumerGroup;
        this.consumeHandler = consumeHandler;
    }

    /**
     * 消费
     *
     * @param message 收到的消息
     */
    @Override
    public void consume(MqMessageReceived message) throws Exception {
        consumeHandler.consume(message);
    }
}

File: folkmq-lang-python/folkmq/folkmq/client/MqSubscription.py
from typing import Callable

from folkmq.client.MqMessageReceived import MqMessageReceived
from folkmq.common.MqConstants import MqConstants


class MqSubscription:
    def __init__(self, topic: str, consumerGroup: str, autoAck: bool, consumeHandler:Callable[[MqMessageReceived], None]):
        self._topic = topic
        self._consumerGroup = consumerGroup
        self._autoAck = autoAck
        self._queueName = topic + MqConstants.SEPARATOR_TOPIC_CONSUMER_GROUP + consumerGroup
        self._consumeHandler = consumeHandler

    def get_topic(self) -> str:
        return self._topic

    def get_consumer_group(self) -> str:
        return self._consumerGroup

    def is_auto_ack(self) -> bool:
        return self._autoAck

    def get_queue_name(self) -> str:
        return self._queueName

    def consume(self, message:MqMessageReceived):
        self._consumeHandler(message)